elastic/kibana
# [Vega] user should be able to set a specific tilemap service using the mapStyle property

## Summary
This is a PR continued discussion of the next comment

> **to discuss:** in accordance with the following documentation https://www.elastic.co/guide/en/kibana/6.8/vega-with-a-map.html `mapStyle` param accepts just a  2 values: `false` and `default`. 
> 
> It looks a little odd to me because in some cases it might be useful to let the user set a specific tms layer value.  For example, if I set `map.tilemap.url` param and want to create some vis with that layer and some with the default one (`road_map`) 
> 
> ```
> {
>   $schema: https://vega.github.io/schema/vega/v5.json
>   config: {
>     kibana: {
>       type: map
>       mapStyle: road_map
>     }
>   }
> }
> ```
> 
> To add support of that we just need to remove the following `if` statement from `vis_type_vega/public/data_model/vega_parser.ts`
> 
>   ```
>   if (res.mapStyle !== `default` && res.mapStyle !== false) {
>       this._onWarning(
>         i18n.translate('visTypeVega.vegaParser.mapStyleValueTypeWarningMessage', {
>           defaultMessage:
>             '{mapStyleConfigName} may either be {mapStyleConfigFirstAllowedValue} or {mapStyleConfigSecondAllowedValue}',
>           values: {
>             mapStyleConfigName: 'config.kibana.mapStyle',
>             mapStyleConfigFirstAllowedValue: 'false',
>             mapStyleConfigSecondAllowedValue: '"default"',
>           },
>         })
>       );
>       res.mapStyle = `default`;
>     }
> ```
> 
> @nreese @thomasneirynck from your point of view should we support that case or not?

_Originally posted by @alexwizp in https://github.com/elastic/kibana/pull/88316#discussion_r557375650_

### For people who will check these changes:
To test that changes please follow next steps: 
1) Start your `Kibana` instance with the following configuration property: 
`map.tilemap.url: http://c.tile.stamen.com/watercolor/{z}/{x}/{y}.jpg`
2) Create or open Vega visualizations with configured Map layer. 
You can use the following spec for testing:
```
{
  $schema: https://vega.github.io/schema/vega/v5.json
  config: {
    kibana: {
      type: map
    }
  }
}
```
3) Apply changes and see which tile map layer was applied.  
Expected screen: 
![image](https://user-images.githubusercontent.com/20072247/104711068-ef049100-5731-11eb-8380-e919963389aa.png)

4) Set mapStyle option to false 
Expected screen:
![image](https://user-images.githubusercontent.com/20072247/104711230-21ae8980-5732-11eb-8bb7-3f810a3b5065.png)

5) Set emsTileServiceId option to `dark_map`. It's one of our default layers. Be sure that `map.includeElasticMapsService` configuration property is true
Expected screen:
![image](https://user-images.githubusercontent.com/20072247/106888755-0fec5080-66f8-11eb-8867-3ec250bc2a81.png)
